integration/pull_test.go

```diff
@@ -796,8 +794,43 @@ min_layer_size=` + minLayerSize + `
 
 	sh.X("soci", "image", "rpull", "--user", regConfig.creds(), "--soci-index-digest", indexDigest, regConfig.mirror(containerImage).ref)
 
-	// default index creation min layer size of 10MB would produce 3 fuse mounts
-	// specified index creation min layer size of 0 would produce 10 fuse mounts
-	// specified runtime min layer size of 25MB will produce 2 fuse mounts
-	checkFuseMounts(t, sh, 2)
+	var snapshotSizes []int64
+	snapshotListOutput := string(sh.O("ctr", "snapshot", "list"))
+	snapshotList := strings.Split(snapshotListOutput, "\n")
+	for _, snapshotLine := range snapshotList[1 : len(snapshotList)-1] {
+		snapshotDigest := strings.Split(snapshotLine, " ")[0]
+		rawJSON := sh.O("ctr", "snapshot", "--snapshotter", "soci", "info", snapshotDigest)
+		var snapshotInfo snapshots.Info
+		if err := json.Unmarshal(rawJSON, &snapshotInfo); err != nil {
+			t.Fatalf("invalid snapshot info from digest %s: %v", snapshotDigest, rawJSON)
+		}
+		strVal := snapshotInfo.Labels[source.TargetSizeLabel]
```

Comment:
We probably should check the `TargetSociIndexDigestLabel` label and skip those not belonging to `indexDigest` to make the test more robust, just in case there is any leftover from other tests or parallel tests.

Reply:
Integration tests run in isolated containers, so that shouldn't be a concern. I could add the check, but don't want to unnecessarily complicate the test.

Reply:
I think it still worth adding the check (given the suggestion below and if it's possible to pull 2 or more images within a single test).

I'd suggest extract the logic (given an image/index, calculate median layer size) to a separate helper func so it can be reused like `indexDigest := buildIndex(sh, imgInfo, withMinLayerSize(medianLayerSize(img)))` to test sparse index. Then we can avoid code that uses fixed imageManifestDigest to ensure sparse index.

https://github.com/awslabs/soci-snapshotter/blob/f3e07639fef7aa7542fad3480dbcbba5eb23e804/integration/pull_test.go#L154-L160

Reply:
The snapshot labels are populated from the manifest. In SOCI we don't have access to the manifest (which is why we have labels), but in integ tests we do.

It might be easier to get the sizes from the manifest like this pull test example:
https://github.com/awslabs/soci-snapshotter/blob/main/integration/pull_test.go#L174-L182

We'd have to pin a manifest version, but I think that's fine.